src/Security/DiscordAuthenticator.php

```diff
@@ -76,16 +76,7 @@ public function getUser($credentials, UserProviderInterface $userProvider): ?Use
             return $user;
         }
 
-        $user = new User();
-        $user->setEmail($discordUser->getEmail());
-        $user->setDisplayName($discordUser->getUsername());
-        $user->setDiscordId($discordUser->getId());
-        $user->setDiscord($discordUser->getUsername().'#'.$discordUser->getDiscriminator());
-        $user->setRoles([User::ROLE_DEFAULT]);
-        $this->entityManager->persist($user);
-        $this->entityManager->flush();
-
-        return $user;
+        return null;
```

Comment:
This should be the only change necessary to prevent users to login via Discord without having an admin user registered in the user table. Then you would have to create an admin user before the user would be able to log in.
If you would like to leave the user creation as it was before, I can revert these changes and remove the user creation.